src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -181,11 +181,25 @@ public void updateFilteredContactList(Predicate<Contact> predicate) {
     }
 
     @Override
-    public void updateFilteredEventList(Predicate<Event> predicate) {
+    public void updateFilteredEventList(Predicate<? super Event> predicate) {
         requireNonNull(predicate);
         filteredEvents.setPredicate(predicate);
     }
 
+    @Override
+    public void sortUpcomingFilteredEventList() {
+        Predicate<? super Event> currentPredicate = filteredEvents
+            .getPredicate();
+        // Remove events that have passed
+        addressBook.sortEvents();
+        // In addition to the current predicate, only include events with end time that is now or after now.
+        // If event has no end time then only include if the start time is now or after now.
+        updateFilteredEventList(event -> (currentPredicate == null || currentPredicate.test(event))
+            && ((event.getEndDateAndTime() == null && event.getStartDateAndTime().isNotBeforeNow())
+                || (event.getEndDateAndTime() != null && event.getEndDateAndTime().isNotBeforeNow()))
```

Comment:
Maybe can separate this to make them look more readable. I feel that it has a different abstraction from `addressBook.sortEvents` and would be better if the logic of the predicate is put elsewhere?